app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -3,8 +3,46 @@ kind: registration
 name: Global Registration
 model: ProvisioningTemplate
 -%>
+#!/bin/sh
+<%
+    curl_headers = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+    curl_vars = { headers: curl_headers, organization: @organization, location: @location, hostgroup: @hostgroup }
+%>
+
+# Template parameters:
+<%= "#  - User: [#{@user.login}]" if @user %>
+<%= "#  - Organization: [#{@organization.name}]" if @organization %>
+<%= "#  - Location: [#{@location.name}]" if @location %>
+<%= "#  - Hostgroup: [#{@hostgroup.name}]" if @hostgroup %>
 
 if ! [ $(id -u) = 0 ]; then
     echo "Please run as root"
     exit 1
 fi
+
+if [ -f /etc/os-release ] ; then
+  . /etc/os-release
+fi
+
+echo "#"
+echo "# Running registration"
+echo "# Hostname: $HOSTNAME"
+echo "# OS: $NAME"
+echo "#"
+
+if [ $ID = rhel ] ; then
```

Comment:
This will fail if `/etc/os-release` is not defined and `$ID` is empty.

```console
$ if [ $ID = rhel ] ; then echo bla ; fi
bash: [: =: unary operator expected
```

The common workaround in POSIX shell is:
```suggestion
if [ x$ID = xrhel ] ; then
```

Also note that you are only compatible with RHEL this way since CentOS presents itself this way:
```console
# grep ^ID /etc/os-release
ID="centos"
ID_LIKE="rhel fedora"
```

subscription-manager is also available on Fedora, but that doesn't present itself as RHEL like:
```console
# grep ^ID /etc/os-release 
ID=fedora
```

Reply:
> This will fail if /etc/os-release is not defined 
I'll check for presence of the file

> Also note that you are only compatible with RHEL
That's okay, we want to use subman only for RHEL systems.

Reply:
> I'll check for presence of the file

You do that, so that part works. Then here you assume it was present and can use the variables. That doesn't work.

> That's okay, we want to use subman only for RHEL systems.

A lot of users don't have RHEL but use CentOS. It'll also be harder to verify it in upstream where we can't test with RHEL. Testing on Fedora also allows testing with newer subman versions. This can make it easier to prepare for the next RHEL version. Given how slow we were with adapting RHEL 8, I'd strongly encourage making it easier to test with the upstream distro.